homeassistant/components/tplink/entity.py

```diff
@@ -311,33 +327,49 @@ def _category_for_feature(feature: Feature | None) -> EntityCategory | None:
 def _description_for_feature[_D: EntityDescription](
     desc_cls: type[_D],
     feature: Feature,
+    descriptions: Mapping[str, _D],
+    *,
+    child_alias: str | None = None,
 ) -> _D:
     """Return description object for the given feature.
 
     This is responsible for setting the common parameters & deciding based on feature id
     which additional parameters are passed.
     """
 
-    kwargs: dict = {}
-    # Key can be overridden for legacy support
-    if (entity_extras := ENTITY_EXTRAS.get(feature.id)) and (
-        key := entity_extras.get("key")
-    ):
-        kwargs["key"] = key
-    else:
-        kwargs["key"] = feature.id
-    # Use translation keys for all features configured in extras
-    if entity_extras:
-        kwargs["translation_key"] = feature.id
-        kwargs["state_class"] = entity_extras.get("state_class")
-        kwargs["device_class"] = entity_extras.get("device_class")
+    enabled_default = feature.category is not Feature.Category.Debug
+    category = _category_for_feature(feature)
+    if descriptions and (desc := descriptions.get(feature.id)):
+        # The state feature gets the device name or the child device
+        # name if it's a child device
+        if feature.id == PRIMARY_STATE_ID:
+            translation_key = None
+            entity_name: str | None | UndefinedType = child_alias
+        else:
+            translation_key = feature.id
+            entity_name = UNDEFINED
```

Comment:
We could set these directly on top & overwrite them for `state` to avoid if-else?

Reply:
Done in https://github.com/home-assistant/core/pull/120064